### PR TITLE
feat: support markdown resources

### PR DESCRIPTION
**Pull request by Xunzhuo**
Added support for markdown.

Signed-off-by: bitliu <bitliu@tencent.com>

### CODE REVIEW

File: docs/design/config-api.md [hunk 1]
```diff
@@ -1,17 +1,6 @@
...
-1. [Motivation](#motivation)
-2. [Goals](#goals)
-3. [Non-Goals](#non-goals)
-4. [Control Plane API](#control_plane_api)
-   1. [Gateway Type](#gateway)
-   2. [Provider Type](#provider)
-   3. [Configuration Examples](#control_plane_configuration)
-6. [Data Plane API](#data_plane_api)
-   1. [Configuration Examples](#data_plane_configuration)
-
```
Comment (danehans) [inline docs/design/config-api.md:14]: Why remove the TOC?

File: docs/user/QUICKSTART.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-## Introduction
+# Quick Start
```
Comment (danehans) [inline docs/user/QUICKSTART.md:1]: Let's keep the name `Quickstart` to stay consistent.

cref: https://en.wikipedia.org/wiki/Quickstart_guide

File: docs/design/watching.md [hunk 3]
```diff
@@ -102,7 +102,7 @@ entries for those pre-existing items; if you are working with `snapshot.Update`
 must add special handling for your first read.  We have a utility function `./internal/message.HandleSubscription` to
 help with this.
 
-### other notes
+### Other notes
```
Comment (danehans) [inline docs/design/watching.md:105]: Since it's a heading, please also capitalize `notes`.

File: docs/design/watching.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-## Watching the results of other components
+## Watching Components Design
```
Comment (danehans) [inline docs/design/watching.md:1]: Can you also update the heading size from `##` to `#`?

File: docs/design/watching.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ To facilitate this communication use the [watchable][] library.  The `watchable.
 standard library's `sync.Map` type, but supports a `.Subscribe` (and `.SubscribeSubset`) method that promotes a pub/sub
 pattern.
 
-### pub
+### Pub
```
Comment (danehans) [inline docs/design/watching.md:12]: Can you also update the heading size from `##`# to `##`?

File: docs/design/watching.md [hunk 6]
```diff
@@ -34,7 +34,7 @@ method) the current value is a no-op; it won't trigger an event for subscribers.
 doesn't have as much state to keep track of; it doesn't need to know "did I already publish this thing", it can just
 `.Store` its data and `watchable` will do the right thing.
 
-### sub
+### Sub
```
Comment (danehans) [inline docs/design/watching.md:37]: Can you also update the heading size from `##`# to `##`?

File: docs/design/watching.md (hunk 3 above)
Comment (danehans) [reply docs/design/watching.md:105]: Can you also update the heading size from `##`# to `##`?

**Review by danehans — CHANGES_REQUESTED** [review]
A few nits throughout. Otherwise LGTM.

**Comment by danehans** [comment]
cc: @kflynn

File: docs/design/config-api.md (hunk 1 above)
Comment (Xunzhuo) [reply docs/design/config-api.md:14]: This will be shown as the title of this post

**Comment by Xunzhuo** [comment]
Thanks for your comments, I have updated them all, and fixed the lint errors along with it.

**Review by danehans — APPROVED** [review]
This puts the docs for the v0.2.0 release in good shape. Thank you!

**Comment by danehans** [comment]
cc: @kflynn

**Pull request merged**